### PR TITLE
Theory and UG for the `SurfaceKinetics` BC

### PR DESCRIPTION
**Pull request by KulaginVladimir**
## Proposed changes

Following #798, this PR adds documentation on the `SurfaceKinetics` BC to Theory and UG sections of docs.

To-do:
- [x] Theory section
- [x] UG section

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [ ] Black formatted
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by KulaginVladimir** [comment]
@RemDelaporteMathurin, I think this can be used as a starting point.

There are two points that I'd like to discuss:
1. Theory section: what do you think of it? Shall we extend it?
2. Shall we note in UG that `AdsorbedHydrogen` was designed specifically for `SurfaceKinetics`?

**Comment by RemDelaporteMathurin** [comment]
I think we also need @ehodille 's review on this one

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:281]: We should specify the dimension of $c_\mathrm{s}$ since it differs from the bulk concentration

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:285]: Don't you want to use thumbnail like in https://github.com/festim-dev/FESTIM/blob/a2fd832f83b952c812b2e8abc2c63d8297ce9261/docs/source/userguide/mesh.rst?plain=1#L89-L91

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:290]: This looks duplicated, do we need both?

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:302]: Well in theory, you could use this model in 2D or 3D if the diffusion is zero. For example if you have an homogenous J_vs on the surface ?

At the moment it's a technical limitation. The steady state version of it #800 could be applied to multi dimension problems

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:338]: You could site papers where it's omitted

File: docs/source/userguide/boundary_conditions.rst
Comment (RemDelaporteMathurin) [inline docs/source/userguide/boundary_conditions.rst:112]: You can refer directly to the right subsection.

```suggestion
Refer to the :ref:`Kinetic surface model` section for more details.
```

File: docs/source/userguide/boundary_conditions.rst
Comment (RemDelaporteMathurin) [inline docs/source/userguide/boundary_conditions.rst:132]:
```suggestion
    my_bc = SurfaceKinetics(
        k_bs=k_bs,
        k_sb=k_sb,
        lambda_IS=1.1e-10,
        n_surf=1e17,
        n_IS=6.3e28,
        J_vs=J_vs,
        surfaces=3,
        initial_condition=0,
        t=t
        )
```

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
This is a very good starting point. I would specify that the AdsorbedConcentration is only for SurfaceKinetics yes.

File: docs/source/theory.rst
Comment (KulaginVladimir) [reply docs/source/theory.rst:302]: In theory, yes. However, I think it can't be used if there is any inhomogeneity (e.g., temperature distribution over the surface, solute source distribution). 

Overall, I wanted to highlight that currently it's 1D+0D. Maybe, we can keep only the part about: The current model does not account for possible surface diffusion?

File: docs/source/theory.rst
Comment (KulaginVladimir) [reply docs/source/theory.rst:290]: Nope

File: docs/source/theory.rst
Comment (KulaginVladimir) [reply docs/source/theory.rst:338]: Meanwhile, I omitted a bunch of papers and kept the minimally needed (in my opinion). Is it OK? There are far more than those I included.

File: docs/source/theory.rst
Comment (KulaginVladimir) [reply docs/source/theory.rst:281]: The units for parameters are not listed at all. Possibly, it'd beneficial to add units to all quantities. What do you think?

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:302]: You are probably right. That's of course if surface diffusion is faster than the other processes.

We want to highlithg that it's only available for 1D problems in FESTIM. We also want to keep that the current model doesn't account for potential surface diffusion 👍

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:338]: Yes of course this is not a paper :-)

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:281]: Yes I think readers would look for units here.

File: docs/source/theory.rst
Comment (KulaginVladimir) [reply docs/source/theory.rst:302]: > We want to highlithg that it's only available for 1D problems in FESTIM.

Shall we add a `warning`? Though we have warnings in API.

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:317]: Actually that's not completely correct. Consider the domain $\Omega = [-1, 0]$ then at $x=0$ the outward normal is $+\vec{x}$.

I know that most people assume positive x values only but is it worth adding something?

File: docs/source/theory.rst
Comment (RemDelaporteMathurin) [inline docs/source/theory.rst:335]:
```suggestion
Usually, these rate constants are expressed in the Arrhenius form: :math:`k_i=k_{i,0}\exp(-E_{k,i} / kT)`. Both these processes are assumed to take place
```

File: docs/source/theory.rst
Comment (KulaginVladimir) [reply docs/source/theory.rst:317]: What do you think of: `At steady-state, eq.(28) for the left domain boundary is reduced to ...`

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:317]: Yes that sounds good. Maybe to be sure we can add "at the left boundary, the normal vector $\mathbf{n}$ is $-\vec{x}$ therefore ..." or something like that?

Comment (KulaginVladimir) [reply docs/source/theory.rst:317]: So, maybe this will fulfill all the gaps?

At the left boundary, the normal vector $\textbf{n}$ is $-\vec{x}$. The steady-state approximation of eq. (8) at the left boundary is, therefore, $D\frac{\partial c_\mathrm{m}}{\partial x}=J_{\mathrm{bs}}-J_{\mathrm{sb}}$

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:317]: perfect!

**Review by RemDelaporteMathurin — APPROVED** [review]
This looks good to me! 
I'll let @ehodille add his review.

File: docs/source/theory.rst
Comment (ehodille) [inline docs/source/theory.rst:284]: Vladimir, do you not want to also mention your papers ?  I think you implemented and used this kind of model in you laser desorption papers (but it is up to you).

Comment (KulaginVladimir) [reply docs/source/theory.rst:284]: @ehodille, well, I used, yes. However, I intended to mention here the pioneers, whereas my papers on LID were mainly based on your works.

File: docs/source/theory.rst
Comment (ehodille) [inline docs/source/theory.rst:303]: does J_vs contains the desorbing flux and all abstraction fluxes (ELy Rideal recombination etc) ?

Comment (KulaginVladimir) [reply docs/source/theory.rst:303]: In general, yes, but it should be defined by a user. Here is an example from [one](https://github.com/KulaginVladimir/FESTIM-SurfaceKinetics-Validation/blob/main/D_damagedW/D_damagedW.ipynb) validation case on how it can be defined:
```python
def J_vs(T, surf_conc, t):
    phi_atom = SP * Gamma_atom * (1 - surf_conc / n_surf) # absorption
    phi_exc = Gamma_atom * sigma_exc * surf_conc # abstraction
    phi_des = 2 * nu0 * (lambda_des * surf_conc) ** 2 * f.exp(-2 * E_des / F.k_B / T) # desorption
    return phi_atom - phi_exc - phi_des
```

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:303]: Maybe this could be explicited then? something like

```
is the net flux of hydrogen atoms from the vacuum onto the surface (eg recombination, dissociation, Eley-Rideal recombination, etc.).
```

Comment (KulaginVladimir) [reply docs/source/theory.rst:303]: In this case, should we somehow clarify that the flux comes **onto the surface**: `J_vs = +Dissoc. - Recomb. - Eley-Rideal recomb ...`?

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:303]: Probably

Comment (KulaginVladimir) [reply docs/source/theory.rst:303]: What about this:

Assuming that there are $N$ processes leading to adsorption of hydrogen with flux $J_{\mathrm{ads},i}$ (e.g. dissociation from the gase phase, adsorption of low-energy ions, etc) and $M$ processes leading to desorption of hydrogen with flux $J_{\mathrm{des},j}$ (e.g. molecular recombindation, Eley-Rideal recombination, etc), the resulting flux onto the surface is: $J_\mathrm{vs}=\sum\limits_i^N J_{\mathrm{ads},i}-\sum\limits_j^M J_{\mathrm{des},j}$

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:303]: Maybe just say J_vs = J_in - J_out where J_in is the sum of all fluxes coming from the vacuum to the surface and J_out from the surface to the vacuum

Or something like that and then give a few examples

Comment (KulaginVladimir) [reply docs/source/theory.rst:303]: A few examples on different processes that can be included?

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst:303]: Yes just like you had in your previous comment

Comment (KulaginVladimir) [reply docs/source/theory.rst]: I updated this part. Let me know what you think.

Comment (RemDelaporteMathurin) [reply docs/source/theory.rst]: Looks good to me!

**Comment by RemDelaporteMathurin** [comment]
@KulaginVladimir I think we can merge this unless @ehodille has something else to add

**Pull request merged**
